ScanbotSDKExamples/MAUI/ClassicComponent.Maui/Platforms/iOS/CustomViews/BarcodeCameraView_iOS.cs

```diff
@@ -0,0 +1,152 @@
+﻿using BarcodeSDK.MAUI.Models;
+using ClassicComponent.Maui.CustomViews;
+using CoreGraphics;
+using DocumentSDK.MAUI.iOS;
+using ScanbotSDK.iOS;
+using UIKit;
+
+namespace ClassicComponent.Maui.Platforms.iOS.CustomViews
+{
+    public class BarcodeCameraView_iOS : UIView
+	{
+        BarcodeCameraViewHandler barcodeCameraViewHandler;
+        SBSDKBarcodeScannerViewController cameraViewController;
+        public BarcodeCameraView_iOS(CGRect frame) : base(frame) { }
+
+        internal async void ConnectHandler(BarcodeCameraViewHandler barcodeCameraViewHandler)
+        {
+            this.barcodeCameraViewHandler = barcodeCameraViewHandler;
+            var visibleViewController = await CheckViewControllerAvailable(barcodeCameraViewHandler);
+            if (visibleViewController != null)
+            {
+                cameraViewController = new SBSDKBarcodeScannerViewController(visibleViewController, barcodeCameraViewHandler.PlatformView);
+                cameraViewController.Delegate = new BarcodeScannerDelegate
+                {
+                    OnDetect = HandleBarcodeScannerResults
+                };
+                cameraViewController.BarcodeImageGenerationType = SBSDKBarcodeImageGenerationType.None;
+                SetConfigurations();
+            }
+        }
+
+        private async Task<UIViewController> CheckViewControllerAvailable(BarcodeCameraViewHandler barcodeCameraViewHandler)
+        {
+            var retryCount = 5;
+
+            UIKit.UIViewController viewController = null;
+            do
+            {
+                await Task.Delay(500);
+                viewController = AppDelegate.ExtractViewController(barcodeCameraViewHandler?.PlatformView?.Window);
+                var message = viewController != null ? "available" : "unavailable";
+                System.Diagnostics.Debug.WriteLine("ViewController is " + message);
+                retryCount++;
+            } while (viewController == null || retryCount < 5);
+            return viewController;
+        }
+
+        #region Properties Implementation
+
+        internal void MapIsFlashEnabled(bool isFlashEnabled)
+        {
+            if (cameraViewController == null) return;
+            cameraViewController.FlashLightEnabled = isFlashEnabled;
+        }
```

Comment:
Do we need a whole method for this?

Reply:
Yes, as discussed on call, it was implemented this way in the public MAUI examples.